src/ApiService/ApiService/onefuzzlib/Auth.cs

```diff
@@ -1,45 +1,65 @@
 ﻿namespace Microsoft.OneFuzz.Service;
-
-using System.Buffers.Binary;
 using System.Diagnostics;
-using System.Security.Cryptography;
+using System.IO;
 
 public class Auth {
 
-    private static ReadOnlySpan<byte> SSHRSABytes => new byte[] { (byte)'s', (byte)'s', (byte)'h', (byte)'-', (byte)'r', (byte)'s', (byte)'a' };
+    private static ProcessStartInfo SshKeyGenProcConfig(string tempFile) {
 
-    private static byte[] BuildPublicKey(RSA rsa) {
-        static Span<byte> WriteLengthPrefixedBytes(ReadOnlySpan<byte> src, Span<byte> dest) {
-            BinaryPrimitives.WriteInt32BigEndian(dest, src.Length);
-            dest = dest[sizeof(int)..];
-            src.CopyTo(dest);
-            return dest[src.Length..];
+        string keyGen = "ssh-keygen";
+        var winAzureFunctionPath = "C:\\Program Files\\Git\\usr\\bin\\ssh-keygen.exe";
+        if (File.Exists(winAzureFunctionPath)) {
+            keyGen = winAzureFunctionPath;
         }
+        var p = new ProcessStartInfo() {
+            FileName = keyGen,
+            CreateNoWindow = false,
+            UseShellExecute = false,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true
+        };
+        p.ArgumentList.Add("-t");
+        p.ArgumentList.Add("rsa");
+        p.ArgumentList.Add("-f");
+        p.ArgumentList.Add(tempFile);
+        p.ArgumentList.Add("-P");
+        p.ArgumentList.Add("");
+        p.ArgumentList.Add("-b");
+        p.ArgumentList.Add("2048");
+        return p;
+    }
 
-        var parameters = rsa.ExportParameters(includePrivateParameters: false);
-
-        // public key format is "ssh-rsa", exponent, modulus, all written
-        // as (big-endian) length-prefixed bytes
-        var result = new byte[sizeof(int) + SSHRSABytes.Length + sizeof(int) + parameters.Modulus!.Length + sizeof(int) + parameters.Exponent!.Length];
-        var spanResult = result.AsSpan();
-        spanResult = WriteLengthPrefixedBytes(SSHRSABytes, spanResult);
-        spanResult = WriteLengthPrefixedBytes(parameters.Exponent, spanResult);
-        spanResult = WriteLengthPrefixedBytes(parameters.Modulus, spanResult);
-        Debug.Assert(spanResult.Length == 0);
+    // This works both on Windows and Linux azure function hosts
+    private static async Async.Task<(string, string)> GenerateKeyValuePair() {
+        var tmpFile = Path.GetTempFileName();
+        File.Delete(tmpFile);
+        tmpFile = tmpFile + ".key";
+        var startInfo = SshKeyGenProcConfig(tmpFile);
+        var proc = new Process() { StartInfo = startInfo };
+        if (proc is null) {
+            throw new Exception($"ssh-keygen process could start (got null)");
+        }
+        if (proc.Start()) {
+            await proc.WaitForExitAsync();
+            if (proc.ExitCode != 0) {
+                throw new Exception($"ssh-keygen failed due to {await proc.StandardError.ReadToEndAsync()}");
```

Comment:
We need to start reading redirected outputs before waiting for exit or the program can get stuck and deadlock (output pipe fills up).

Reply:
Set `RedirectStandardOutput = false` since we never read it, then put this read before the `WaitForExitAsync`.